### PR TITLE
Run training jobs without ssh in the local machine.

### PR DESCRIPTION
**Pull request by zheng-da**
*Description of changes:*
When running a training job on the local machine, we can avoid using ssh.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by zhjwy9343** [comment]
After we change this behavior, should we update any documentations?

File: python/graphstorm/run/launch.py [hunk 1]
```diff
@@ -733,8 +913,56 @@ def signal_handler(sig, frame): # pylint: disable=unused-argument
...
+                    ip = result[0]
+                    port = get_available_port(ip)
+                    hosts.append((ip, port))
+                else:
+                    raise RuntimeError("Format error of ip_config.")
+    else:
+        # The user doesn't provide an IP config file. This means we are going to
+        # run the training job in the local machine. We should create a temporary
+        # IP config file.
+        with tempfile.NamedTemporaryFile(delete=False) as fp:
```
Comment (classicsong) [inline python/graphstorm/run/launch.py:957]: Who will remove this file?

Comment (zhjwy9343) [reply python/graphstorm/run/launch.py:957]: And why do we still need this temp file as we already know the ip address?

Comment (zheng-da) [reply python/graphstorm/run/launch.py:957]: DistDGL needs this file.

Comment (zheng-da) [reply python/graphstorm/run/launch.py:957]: right now it's not deleted. this is a temporary file. should we delete it?

File: docs/source/install/env-setup.rst [hunk 2]
```diff
@@ -47,7 +47,9 @@ For CPU environment:
 
 Configure SSH No-password login
 ................................
-Use the following commands to configure a local SSH no-password login that GraphStorm relies on.
+To perform distributed training in a cluster of machines, please use the following commands
+to configure a local SSH no-password login that GraphStorm relies on.
+*Note*: this is not needed for the standalone mode.
```
Comment (zhjwy9343) [inline docs/source/install/env-setup.rst:52]: Can you use the blow rst code to highlight this Node.

.. note::

    This operation is not needed for the standalone mode.

File: docs/source/tutorials/quick-start.rst [hunk 3]
```diff
@@ -12,12 +12,6 @@ GraphStorm is designed for easy-to-use GML models, particularly the graph neural
 
 This tutorial will use GraphStorm's built-in OGB-arxiv dataset for a node classification task to demonstrate these three steps.
 
-.. warning::
```
Comment (zhjwy9343) [inline docs/source/tutorials/quick-start.rst:15]: Could you also help to remove the same warning section in the `own-data.rst` line 11-15?

**Review by zhjwy9343 — APPROVED** [review]
LGTM

**Pull request merged**